server.go

```diff
@@ -149,6 +150,30 @@ func CustomCodec(codec Codec) ServerOption {
 	}
 }
 
+// CustomCodecs returns a ServerOption that sets the codecs that can be used based on the
+// content-type subtype, ie if a codec returns "proto" from calling String(), this will
+// be used if the content-type is "application/grpc+proto". If there is no subtype
+// on the content-type, the codec set by CustomCodec will be used, or if CustomCodec
+// is not set, then the default codec will be used.
+func CustomCodecs(codecs ...Codec) ServerOption {
+	m := make(map[string]Codec, len(codecs))
+	for _, codec := range codecs {
+		// this is a one-time server setup operation
+		// note that http/2 transmits headers as lowercase
+		name := strings.TrimSpace(strings.ToLower(codec.String()))
```

Comment:
if you add a `codec.ContentSubtype` method as proposed above, you will have symmetry in the client and server code paths.